dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/zk/CuratorZookeeperClient.java

```diff
@@ -0,0 +1,119 @@
+import org.apache.curator.framework.api.ACLProvider;
+import org.apache.curator.framework.state.ConnectionState;
+import org.apache.curator.retry.ExponentialBackoffRetry;
+import org.apache.zookeeper.ZooDefs;
+import org.apache.zookeeper.data.ACL;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+import static org.apache.dolphinscheduler.common.utils.Preconditions.checkNotNull;
+
+/**
+ * Shared Curator zookeeper client
+ */
+@Component
+public class CuratorZookeeperClient implements InitializingBean {
+    private final Logger logger = LoggerFactory.getLogger(CuratorZookeeperClient.class);
+
+    @Autowired
+    private ZookeeperConfig zookeeperConfig;
+
+    private CuratorFramework zkClient;
+
+
+    @Override
+    public void afterPropertiesSet() throws Exception {
+        this.zkClient = buildClient();
+        initStateLister();
+    }
+
+    private CuratorFramework buildClient() {
+        logger.info("zookeeper registry center init, server lists is: {}.", zookeeperConfig.getServerList());
+
+        CuratorFrameworkFactory.Builder builder = CuratorFrameworkFactory.builder().ensembleProvider(new DefaultEnsembleProvider(checkNotNull(zookeeperConfig.getServerList(),"zookeeper quorum can't be null")))
+                .retryPolicy(new ExponentialBackoffRetry(zookeeperConfig.getBaseSleepTimeMs(), zookeeperConfig.getMaxRetries(), zookeeperConfig.getMaxSleepMs()));
+
+        //these has default value
+        if (0 != zookeeperConfig.getSessionTimeoutMs()) {
+            builder.sessionTimeoutMs(zookeeperConfig.getSessionTimeoutMs());
+        }
+        if (0 != zookeeperConfig.getConnectionTimeoutMs()) {
+            builder.connectionTimeoutMs(zookeeperConfig.getConnectionTimeoutMs());
+        }
+        if (StringUtils.isNotBlank(zookeeperConfig.getDigest())) {
+            builder.authorization("digest", zookeeperConfig.getDigest().getBytes(StandardCharsets.UTF_8)).aclProvider(new ACLProvider() {
+
+                @Override
+                public List<ACL> getDefaultAcl() {
+                    return ZooDefs.Ids.CREATOR_ALL_ACL;
+                }
+
+                @Override
+                public List<ACL> getAclForPath(final String path) {
+                    return ZooDefs.Ids.CREATOR_ALL_ACL;
+                }
+            });
+        }
+        zkClient = builder.build();
+        zkClient.start();
+        try {
+            zkClient.blockUntilConnected();
+        } catch (final Exception ex) {
+            throw new RuntimeException(ex);
+        }
+        return zkClient;
+    }
+
+    public void initStateLister() {
+        checkNotNull(zkClient);
+
+        zkClient.getConnectionStateListenable().addListener((client, newState) -> {
```

Comment:
Hi,
Because remove the `master zk address` log from `MasterRegistry`, please add the `master zk address` to log, not the zk server info.

------
因为我们是从masterregistry中删除了master zk节点的日志，想要添加到CuratorZkCLient中。所以CuratorZkCLient日志信息应该是`master zk节点的日志`信息，而不是zk地址信息。

If you have any question or suggestion, please put forward.

Reply:
你好，原来的日志打印的也是zookeeper的地址。另外，这种连接状态的监控，应该是用一个listener监听就可以了，重复的监听是没有意义的。

Reply:
> 你好，原来的日志打印的也是zookeeper的地址。另外，这种连接状态的监控，应该是用一个listener监听就可以了，重复的监听是没有意义的。

![image](https://user-images.githubusercontent.com/29545877/88020828-f8d08180-cb5e-11ea-8fae-0835eddd9622.png)

Reply:
又看了一下，这里不是日志的问题，而是原来代码里在Reconnected的时候，有个保存临时节点的操作。那可能确实不能简单地删掉，还要再分析下

Reply:
> 又看了一下，这里不是日志的问题，而是原来代码里在Reconnected的时候，有个保存临时节点的操作。那可能确实不能简单地删掉，还要再分析下

I agree with you~

Reply:
已经把对这个类的修改恢复了